packages/devp2p/src/rlpx/peer.ts

```diff
@@ -699,18 +699,15 @@ export class Peer extends EventEmitter {
    * @param msg Message text to debug
    */
   private debug(messageName: string, msg: string, disconnectReason?: string) {
-    debug(msg)
-    if (this.msgDebuggers[messageName]) {
-      this.msgDebuggers[messageName](msg)
-    }
     const ip = this._socket.remoteAddress
-    if (ip && this.msgDebuggers[ip]) {
-      this.msgDebuggers[ip](msg)
-    }
     if (disconnectReason && messageName === 'DISCONNECT') {
       if (this.msgDebuggers[disconnectReason]) {
         this.msgDebuggers[disconnectReason](msg)
       }
-    }
+    } else if (this.msgDebuggers[messageName]) {
```

Comment:
@holgerd77 Which messages did we want to dedup on this one?  With this PR's changes, it looks like this:
![image](https://user-images.githubusercontent.com/17355484/149851907-154c0d3d-5801-402e-9f7a-b3c4a6e76d57.png)

And on current `master`, it looks like this.
![image](https://user-images.githubusercontent.com/17355484/149851999-670621eb-f1a3-46e2-8d92-3172d150755d.png)

Should we be removing the IP based messages?  I believe with these changes we will lose the IP based messages entirely since I don't know if we'll ever encounter a situation where there is an IP based message debugger that doesn't also have a `messageName` debugger (which is a higher priority in the conditional logic here.  Maybe we should change it to something like below so users could specify if they want to see log messages from a specific IP address:
```
if (ip && process.env.DEBUG === `devp2p:${ip}`) {
      this.msgDebuggers[ip](msg)
    } else if (disconnectReason && messageName === 'DISCONNECT') {
      if (this.msgDebuggers[disconnectReason]) {
        this.msgDebuggers[disconnectReason](msg)
      }...
```

It produces output that looks something like this:
![image](https://user-images.githubusercontent.com/17355484/149853232-e5a48416-3f5a-4784-b817-cec69854113e.png)

We'd have to apply this same check everywhere we have the IP based messaging and then add some updates to the docs to reflect this usage but it seems to work well.

Reply:
These `devp2p:127.0.0.1` or `devp2p:rlpx:peer:HELLO` prefixes on the log messages are actually *only* the names of the loggers activated and not a "feature" to display the IP adress on some loggers. 😋

So on your usage this is the correct dedublication, the main point of this whole thing is the we do not have situations where the same message (in the screenshot from above e.g. "Received HELLO message...") is sent multiple times.

And even if only `devp2p:rlpx:peer:HELLO` is logged, the remote peer IP is visible in the debug message itself.

That said:

I also tested a local connection along [these](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/client/examples/local-debugging.md#local-connections--ethereumjs---ethereumjs) instructions with the following command:

```shell
cd ../devp2p && npm run build && cd ../client && DEBUG=devp2p:127.0.0.1 npm run client:start:dev2 -- --bootnodes=enode://90e0424dff457a1b677b45aa0bf861e71284c45514e944060f8fb3cca65ddc1b241e686e7a9fb8ac37e6090376a4faf379368cc8ad59ae29194ee3126c3e9c88@127.0.0.1:30303
```

And with the current conditional order the IP logger wouldn't display in this case.

I guess it's enough if we just switch the meesage logger and the IP logger order and keep the rest in place, so this would work for me locally:

```typescript
const ip = this._socket.remoteAddress
if (disconnectReason && messageName === 'DISCONNECT') {
  if (this.msgDebuggers[disconnectReason]) {
    this.msgDebuggers[disconnectReason](msg)
  }
} else if (ip && this.msgDebuggers[ip]) {
  this.msgDebuggers[ip](msg)
} else if (this.msgDebuggers[messageName]) {
  this.msgDebuggers[messageName](msg)
} else {
  debug(msg)
}
```

Short nit for @ScottyPoi (actually all of this comment is mainly for Scotty of course 😛): it would be good if you also add brackets to one-liner code (so this simple `debug(msg)`. We had some bugs lately caused by someone adding something below such a line and not noticing this is not executed in this conditional context. So this would be a bit more on the safe side.

Note: this order change from above would also need to be adjusted in the `eth/index.ts` file.